pod/perlop.pod

```diff
@@ -2533,9 +2533,9 @@ expression is used instead.  See L<perlre> for further explanation on these.
 Options are as with C<m//> with the addition of the following replacement
 specific options:
 
-    e   Evaluate the right side as an expression.
-    ee  Evaluate the right side as a string then eval the
-        result.
+    e   Evaluate the right side as a block of code.
+    ee  Evaluate the right side as a block of code, then eval()
+        the resulting string.
```

Comment:
Where exactly in the source code does this evaluation of the RHS **as a block of code** rather than as either `an expression` or a `string` is enforced?  (I'm asking this not to quibble, but just to know your reference point.)

Reply:
I didn't know this beforehand. This PR was written from a Perl programmer POV; I just know how `/e` behaves in practice. You can put multiple statements separated by `;` in the replacement part; it acts as a block scope; etc.

But I looked it up. The relevant part of the perl source is `S_scan_subst` in `toke.c`. It has special handling for `EXEC_PAT_MOD` flags (defined as `'e'`), which it keeps a count of in `es`. If `es` is non-zero, it turns on the `PMf_EVAL` flag on the op and literally wraps `do { ... }` around the replacement part. Then, for every `e` beyond the first one, it prepends `eval ` to the replacement part. In other words, it doesn't even act on op trees; it literally generates source code of the form `("eval " x ($es - 1)) . "do { $RHS }"`. I didn't expect that.